### PR TITLE
added manual distance calculation if Google fails

### LINKED ISSUE #35: Backup for Google Distance API
### Feature Description
Need a backup mechanism to fire if our app exceeds the daily request quota for Google's Distance API. Affects `MarkerCardsFact.getDistanceToMarker()` method, potentially `AllMarkersFact` as well.

### Recommended Actions

Use this formula to calculate the "bird's eye" distance to each marker:
```sh
    function calculateDistanceToMarker(lat1, lon1, lat2, lon2){
      var R = 6371e3; // metres
      var φ1 = toRadians(lat1);
      var φ2 = toRadians(lat2);
      var Δφ = toRadians((lat2-lat1));
      var Δλ = toRadians((lon2-lon1));

      var a = Math.sin(Δφ/2) * Math.sin(Δφ/2) +
              Math.cos(φ1) * Math.cos(φ2) *
              Math.sin(Δλ/2) * Math.sin(Δλ/2);
      var c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1-a));
      var d = R * c;
      return getMiles(d).toPrecision(2);
    }

    function toRadians(x) {
       return x * Math.PI / 180;
    }

    // Converts meters to miles
    function getMiles(i) {
     return i*0.000621371192;
    }
```

Source: http://www.movable-type.co.uk/scripts/latlong.html

### PR DESCRIPTION
**Pull request by MattKraatz**
## Status
**READY**

## Documentation

[X] I have fully documented features in this PR's code. Please put an `X` in the box to confirm.

[ ] This PR does **NOT** require documentation.

## Description
Catch Google Maps errors in `MarkerCardsFact.getDistanceToMarker()` and use a manual calculation instead.

## Resolves Issue Number
Resolves #35 

## Deploy Notes

```sh
git pull --prune
git checkout kraatz-manual-distance
ionic serve --lab
```

## Steps to Test or Reproduce
Outline the steps to test or reproduce the PR here.

1. If we've hit our API call limit for the day, make sure we still get distances back.

## Impacted Files in Application
List general components of the application that this PR will affect:

* factories/MarkerCardFact.js
* controllers/MarkersCtrl.js

### CODE REVIEW

**Review by delaineackerman — COMMENTED** [review]
LGTM

**Review by delaineackerman — APPROVED** [review]
LGTM

File: NashvilleHistory/www/js/factories/MarkerCardsFact.js [hunk 1]
```diff
@@ -45,13 +45,45 @@ app.factory("MarkerCardsFact", ($q, $http, KeyGetter)=>{
...
   }
 
+  function calculateDistanceToMarker(lat1, lon1, lat2, lon2){
+    var R = 6371e3; // metres
+    var φ1 = toRadians(lat1);
+    var φ2 = toRadians(lat2);
+    var Δφ = toRadians((lat2-lat1));
+    var Δλ = toRadians((lon2-lon1));
+
+    var a = Math.sin(Δφ/2) * Math.sin(Δφ/2) +
```
Comment (matthamil) [inline NashvilleHistory/www/js/factories/MarkerCardsFact.js:74]: Can you put a comment with the formula? I'm assuming all of this works flawlessly. Not sure how to reproduce and test 100% without maxing out my call limit :smile:. LGTM

**Review by matthamil — APPROVED** [review]
LGTM. Math.

File: NashvilleHistory/www/js/factories/MarkerCardsFact.js (hunk 1 above)
Comment (MattKraatz) [reply NashvilleHistory/www/js/factories/MarkerCardsFact.js:74]: Good call, will add a reference to the source

**Pull request merged**